### DISCUSSION TITLE
logging parameter (x,y,z) when using lighthouse positioning system

### DISCUSSION THREAD

**Discussion by imreburn** [bitcraze/discussions#1027] [Q&A] [answered]
Hi,

With lighthouse positioning system (and no other positioning systems for crazyflies), I would like to log position, velocity, etc. [Here](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/functional-areas/lighthouse/positioning_methods/) says `.config` should be modified, and then position can be logged with `lighthouse.x/.y/.z`. By looking [ICRA 2023 demo](https://github.com/bitcraze/crazyflie-firmware-experimental/tree/icra-2023/examples/demos/decentralized_swarm), it seems it depends on the lighthouse, but it uses `stateEstimate.x/.y/.z` for logging - do these come from elsewhere than lighthouse? What is the meaning of "lighthouse positioning system as groundtruth" in the config?

Another small question: Should `.config` updated via `make menuconfig` whenever the firmware is complied? Is there a way to fix the setting or quicker way to update?

Thanks!

**Comment by krichardsson** [comment, marked as answer]
All you have to do is to log the variables in `stateEstimate.x/.y/.z`. The normal operation is that the data from the lighthouse system (and other sensors) is sent to the estimator to keep track of the state (position, velocity and attitude) and this is what you log from the stateEstimate group.

> lighthouse positioning system as groundtruth

Can be used if you are working with some other positioning system, for instance the Loco system, and want to track the performance of the Loco system using the lighthouse system as a ground truth. In this case the lighthouse data is not sent to the estimator (the Loco data is) and instead a separate positioning process (the crossing beam method) is used for the lighthouse data to provide a completely independent position estimate.

> Another small question: Should .config updated via make menuconfig whenever the firmware is complied? Is there a way to fix the setting or quicker way to update?

When you run `make menuconfig` it will modify the `build/.config` file. You can save this file somewhere and reuse it, or you can even put a copy in the `configs` directory to make it possible to do something like `make my_funky_config`. If you are using an app there is a config in the file tree where you can add your specific settings as well.

**Reply by imreburn** [reply]
Thank you for clarification! I think I don't have to touch `.config` in my case.
